R/learner_mgcv_classif_gam.R

```diff
@@ -0,0 +1,190 @@
+        ParamDbl$new("irls.reg", default = 0.0, lower = 0, tags = c("train", "control")),
+        ParamDbl$new("epsilon", default = 1e-07, lower = 0, tags = c("train", "control")),
+        ParamInt$new("maxit", default = 200L, tags = c("train", "control")),
+        ParamLgl$new("trace", default = FALSE, tags = c("train", "control")),
+        ParamDbl$new("mgcv.tol", default = 1e-07, lower = 0, tags = c("train", "control")),
+        ParamInt$new("mgcv.half", default = 15L, lower = 0L, tags = c("train", "control")),
+        ParamDbl$new(
+          "rank.tol",
+          default = .Machine$double.eps^0.5,
+          lower = 0,
+          tags = c("train", "control")
+        ),
+        ParamUty$new("nlm", default = list(), tags = c("train", "control")),
+        ParamUty$new("optim", default = list(), tags = c("train", "control")),
+        ParamUty$new("newton", default = list(), tags = c("train", "control")),
+        ParamInt$new("outerPIsteps", default = 0L, lower = 0L, tags = c("train", "control")),
+        ParamLgl$new("idLinksBases", default = TRUE, tags = c("train", "control")),
+        ParamLgl$new("scalePenalty", default = TRUE, tags = c("train", "control")),
+        ParamInt$new("efs.lspmax", default = 15L, lower = 0L, tags = c("train", "control")),
+        ParamDbl$new("efs.tol", default = .1, lower = 0, tags = c("train", "control")),
+        ParamFct$new(
+          "scale.est",
+          levels = c("fletcher", "pearson", "deviance"),
+          default = "fletcher",
+          tags = c("train", "control")
+        ),
+        ParamLgl$new("edge.correct", default = FALSE, tags = c("train", "control")),
+        ParamInt$new("block.size", default = 1000L, tags = "predict"),
+        ParamLgl$new("unconditional", default = FALSE, tags = "predict")
+      ))
+
+      super$initialize(
+        id = "classif.gam",
+        packages = "mgcv",
+        feature_types = c("logical", "integer", "numeric"),
+        predict_types = c("prob", "response"),
+        param_set = ps,
+        properties = c("twoclass", "weights"),
+        man = "mlr3extralearners::mlr_learners_classif.gam"
+      )
+    }
+  ),
+
+  private = list(
+    .train = function(task) {
+
+      pars = self$param_set$get_values(tags = "train")
+
+      # set column names to ensure consistency in fit and predict
+      self$state$feature_names = task$feature_names
+
+      data = task$data(cols = c(task$feature_names, task$target_names))
+      if ("weights" %in% task$properties) {
+        pars$weights = task$weights$weight
+      }
+      if (is.null(pars$formula)) {
+        formula = stats::as.formula(paste(
+          task$target_names,
+          "~",
+          paste(task$feature_names, collapse = " + ")
+        ))
+        pars$formula = formula
+      }
+      pars$family = "binomial"
+
+      control_pars = self$param_set$get_values(tags = "control")
+      if (length(control_pars) > 0 ) {
```

Comment:
mlr style removes `>0`

```suggestion
      if (length(control_pars)) {
```